cdap-ui/app/cdap/components/HttpExecutor/RequestHistoryTab/index.tsx

```diff
@@ -0,0 +1,183 @@
+import { connect } from 'react-redux';
+
+export interface IRequestHistory {
+  method: RequestMethod;
+  path: string;
+  body: string;
+  headers: {
+    pairs: [
+      {
+        key: string;
+        value: string;
+        uniqueId: string;
+      }
+    ];
+  };
+  response: string;
+  statusCode: number;
+}
+
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      borderRight: `1px solid ${theme.palette.grey[300]}`,
+      height: '100%',
+    },
+    requestRow: {
+      padding: '10px',
+      lineHeight: '24px',
+      display: 'grid',
+      width: '100%',
+      gridTemplateColumns: '50px 1fr',
+      cursor: 'pointer',
+
+      '&:hover': {
+        backgroundColor: theme.palette.grey[700],
+      },
+    },
+    requestMethod: {
+      paddingLeft: '5px',
+      color: theme.palette.white[50],
+      width: '100%',
+      height: '100%',
+      fontWeight: 600,
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'flex-start',
+      fontSize: '10px',
+    },
+    requestMethodText: {
+      width: '100%',
+      textAlign: 'left',
+      alignSelf: 'center',
+    },
+    requestPath: {
+      maxWidth: '80%',
+      minWidth: '80%',
+      wordWrap: 'break-word',
+      textAlign: 'left',
+      alignSelf: 'center',
+      textTransform: 'lowercase',
+      fontSize: '10px',
+      display: 'inline-block',
+      lineHeight: '1.3',
+    },
+    getMethod: {
+      color: theme.palette.green[50],
+    },
+    postMethod: {
+      color: theme.palette.orange[50],
+    },
+    putMethod: {
+      color: theme.palette.yellow[50],
+    },
+    deleteMethod: {
+      color: theme.palette.red[50],
+    },
+  };
+};
+
+interface IRequestHistoryTabProps extends WithStyles<typeof styles> {
+  requestLog: List<IRequestHistory>;
+  setRequestLog: (requestLog: List<IRequestHistory>) => void;
+  onRequestClick: (request: IRequestHistory) => void;
+}
+
+const mapStateToProps = (state) => {
+  return {
+    requestLog: state.http.requestLog,
+  };
+};
+
+const mapDispatch = (dispatch) => {
+  return {
+    setRequestLog: (requestLog: List<IRequestHistory>) => {
+      dispatch({
+        type: HttpExecutorActions.setRequestLog,
+        payload: {
+          requestLog,
+        },
+      });
+    },
+    onRequestClick: (request: IRequestHistory) => {
+      dispatch({
+        type: HttpExecutorActions.setRequestHistoryView,
+        payload: request,
+      });
+    },
+  };
+};
+
+const RequestHistoryTabView: React.FC<IRequestHistoryTabProps> = ({
+  classes,
+  requestLog,
+  setRequestLog,
+  onRequestClick,
+}) => {
+  // Query through localstorage to populate RequestHistoryTab
+  React.useEffect(() => {
+    const storedLogs = localStorage.getItem(REQUEST_HISTORY);
+    if (storedLogs) {
+      try {
+        setRequestLog(List(JSON.parse(storedLogs)));
+      } catch (e) {
+        setRequestLog(List([]));
+      }
+    } else {
+      localStorage.setItem(REQUEST_HISTORY, JSON.stringify([]));
```

Comment:
nit: I don't think this is needed

Reply:
This is needed for people who are visiting the site for the first time. They don't have REQUEST_HISTORY key set up in their localstorage.